CMakeLists.txt

```diff
@@ -33,8 +33,19 @@ include(FetchContent)
 include(FindPkgConfig)
 include(AvifExternalProjectUtils)
 
+option(AVIF_ENABLE_NODISCARD "Add [[nodiscard]] to some functions. CMake must be at least 3.21 to force C23." OFF)
+
 # Set C99 as the default
-set(CMAKE_C_STANDARD 99)
+if(AVIF_ENABLE_NODISCARD)
+    # [[nodiscard]] requires C23.
+    if(CMAKE_VERSION VERSION_LESS 3.21.0)
+        message(FATAL_ERROR "CMake must be at least 3.21 to force C23, bailing out")
+    endif()
+    set(CMAKE_C_STANDARD 23)
+    set(CMAKE_C_STANDARD_REQUIRED ON)
+else()
+    set(CMAKE_C_STANDARD 99)
```

Comment:
I would think this would be required too, but it's highly unlikely anyone would try this with a c90 only compiler (or force those flags)

Reply:
Yes, we assume the compiler supports C99.